docs/pages/docs/advanced/foundry.mdx

```diff
@@ -0,0 +1,126 @@
+
+Foundry projects follow various development workflows (test-driven, deploy to a fresh chain, deploy to a fork, etc). Rather than a one-size-fits-all integration, this page offers patterns that you can adapt to your workflow.
+
+The Foundry [example project](https://github.com/ponder-sh/ponder/tree/main/examples/with-foundry) includes a simple implementation of each pattern described in this guide.
+
+## Configure the `anvil` network
+
+### Disable caching
+
+Ponder's RPC request cache works well for live networks where the chain is generally immutable, but causes issues when indexing a local chain that "resets".
+
+Use the `disableCache` option to **disable RPC request caching** for the Anvil network. With this option set to true, Ponder will clear the cache on start up and between hot reloads.
+
+```ts filename="ponder.config.ts" {9}
+import { createConfig } from "@ponder/core";
+import { http } from "viem";
+
+export default createConfig({
+  networks: {
+    anvil: { 
+      chainId: 31337,
+      transport: http("http://127.0.0.1:8545"),
+      disableCache: true,
+    },
+  },
+  // ...
+});
+```
+
+### Chain ID
+
+We recommend using `31337` (the default Anvil chain ID) even when forking a live chain. This avoids common footguns when working with multiple networks.
+
+### Mining mode
+
+We recommend using [interval mining](https://book.getfoundry.sh/reference/anvil/#mining-modes) with a block time of ~2 seconds. This better simulates a live network.
+
+<Callout type="warning">Known issue: When indexing Anvil with auto mining enabled in an app with multiple networks, indexing progress will get "stuck" at the timestamp of the latest Anvil block. </Callout>
+
+## Generate ABI files
+
+To enable end-to-end type safety, the contract ABIs generated by Foundry must be copied into TypeScript (`.ts`) source files.
+
+### Wagmi CLI
+
+The Wagmi CLI [Foundry plugin](https://wagmi.sh/cli/api/plugins/foundry) is an excellent tool to automate tedious ABI file management. For more information, visit the [Wagmi CLI documentation](https://wagmi.sh/cli/getting-started).
+
+Here is the Wagmi CLI config file used by the Foundry [example project](https://github.com/ponder-sh/ponder/tree/main/examples/with-foundry).
+
+```ts filename="wagmi.config.ts"
+import { defineConfig } from "@wagmi/cli";
+import { foundry } from "@wagmi/cli/plugins";
+
+export default defineConfig({
+  out: "abis/CounterAbi.ts",
+  plugins: [
+    foundry({
+      project: "foundry",
+      include: ["Counter.sol/**"],
+    }),
+  ],
+});
+```
+
+## Import broadcast files
+
+When you run a Foundry script with [broadcast](https://book.getfoundry.sh/tutorials/solidity-scripting?highlight=deploy#deploying-locally) enabled, Foundry writes transaction inputs and receipts to a JSON file in the `broadcast` directory. You can import these files directly into `ponder.config.ts` to automate address management and enable hot reloading.
```

Comment:
This is a little confusing to me and might be too much detail because without "broadcast" nothing is persisted on-chain so you wouldn't be able to index